### PR TITLE
feat(meta): Add V2 client for adding, updating device profiles

### LINKED ISSUE #386: [V2 meta] Add V2 client for adding, updating device profiles
Add V2 client for adding, updating device profiles
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/post_deviceprofile
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/default/put_deviceprofile

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #386

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Add Put helper method and V2 client for adding, updating device profiles

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/clients/http/deviceprofile.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	v2 "github.com/edgexfoundry/go-mod-core-contracts/v2"
```
Comment (cloudxxx8) [inline v2/clients/http/deviceprofile.go:11]: you don't need the alias here

File: v2/clients/http/deviceprofile.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
```
Comment (cloudxxx8) [inline v2/clients/http/deviceprofile.go:9]: add a blank line

File: v2/clients/http/deviceprofile_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	v2 "github.com/edgexfoundry/go-mod-core-contracts/v2"
```
Comment (cloudxxx8) [inline v2/clients/http/deviceprofile_test.go:6]: add a blank line and remove the alias

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/clients/http/utils/put.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+)
+
+// Helper method to make the put JSON request and return the body
```
Comment (lenny-intel) [inline v2/clients/http/utils/put.go:18]: This is 99% duplicate of POST. Refactor to one that takes the Method as parameter.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/clients/http/deviceprofile.go (hunk 2 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile.go:9]: Added.

File: v2/clients/http/deviceprofile.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile.go]: Modified.

File: v2/clients/http/deviceprofile_test.go (hunk 3 above)
Comment (weichou1229) [reply v2/clients/http/deviceprofile_test.go]: Modified.

File: v2/clients/http/utils/put.go (hunk 4 above)
Comment (weichou1229) [reply v2/clients/http/utils/put.go]: Refactored the help method.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**